bundles/http/Citrix/SAP-Concur-Demo-HTTP/script.js

```diff
@@ -0,0 +1,603 @@
+        ReportName: '12372882'
+        },
+        {
+        Id: '19710704432971788', 
+        Vendor: 'Staples', 
+        SpendingCategory: 'Office Supplies',
+        TransactionDate: date2.toLocaleDateString(), 
+        CashAdvanceApplied: false, 
+        Amount: 200.0000, 
+        CurrencyCode: 'USD', 
+        ReportName: '12372882'
+        },
+        {
+        Id: '19710704432971791', 
+        Vendor: 'ESET', 
+        SpendingCategory: 'Office Software',
+        TransactionDate: date2.toLocaleDateString(), 
+        CashAdvanceApplied: false, 
+        Amount: 600.0000, 
+        CurrencyCode: 'USD', 
+        ReportName: '12372882'
+        },
+        {
+        Id: '19710704732971819', 
+        Vendor: 'Starwood Hotels', 
+        SpendingCategory: 'Lodging',
+        TransactionDate: date3.toLocaleDateString(), 
+        CashAdvanceApplied: false, 
+        Amount: 200.0000, 
+        CurrencyCode: 'USD', 
+        ReportName: '12381535'
+        }
+  ]
+
+const categories = [
+  {
+    Type: 'Ground Transportation',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/meeting.a44eb1e81b027ba58ea71078394ca1a0.svg'
+  },
+  {
+    Type: 'Lodging',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/account.7e6a4a82a354a98226bc0c0b4ec9039b.svg'
+  },
+  {
+    Type: 'Airline Fee',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/itinerary.82ae0728d114f607d5b618e3b4eba614.svg'
+  },
+  {
+    Type: 'Meal',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/change_request.b0ee4a80fb3ec8b889c043706b26bb1d.svg'
+  },
+  {
+    Type: 'Entertainment',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/default-broken.b3426980ecaefcfc0778d475c1fe4dd6.svg'
+  },
+  {
+    Type: 'Other',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/feedback.c533ed2d280d39d9288b2fd741a18651.svg'
+  },
+  {
+    Type: 'Office Supplies',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/case.2bb73f1b3eff52a8134fd591447bdaa1.svg'
+  },
+  {
+    Type: 'Advertising/Marketing',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/lead.e61e72321674c9469b115c0552c5dc65.svg'
+  },
+  {
+    Type: 'Office Software',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/report_update.2d453d1646665c2ede6de83f848a6566.svg'
+  }
+]
+
+async function syncReports ({ dataStore, integrationParameters }) {
+  console.log('Synching data...')
+
+  countTotalAmount()
+
+  dataStore.save('reports', reports)
+  dataStore.save('expenses', expenses)
+  dataStore.save('categories', categories)
+  console.log("Synchronization complete")
+}
+
+function countTotalAmount() {
```

Comment:
try to avoid using globals, especially modifying globals:

```
function calculateReportsWithTotals(expenses, reports) {
  const amounts = expenses.reduce((acc, expense) => {
    if (!acc.has(expense.ReportName) {
      acc.set(expense.ReportName, 0)
    }
    acc.set(expense.ReportName, acc.get(expense.ReportName) + expense.Amount)
    return acc
    }, new Map() )
  return reports.map(report => ({...report, TotalAmount: amounts.get(report.Id)}))
}
```
`Map` is not so popular in Javascript, legacy approach with Object would be more readable for Javascript developers:

```
function calculateReportsWithTotals(expenses, reports) {
  const amounts = expenses.reduce((acc, {ReportName, Amount}) => {    
    acc[ReportName] = (acc[ReportName] || 0) + Amount
    return acc
    }, {} )
  return reports.map(report => ({...report, TotalAmount: amounts.get(report.Id)}))
}
```